### DISCUSSION TITLE
Inline Javascript doesnt work on nested component

### DISCUSSION THREAD

**Discussion by aminulloh2002** [livewire/livewire#6662] [1. Help]
So, i have dynamic modal content that will render if openModal event is fired,
it worked nicely in v2 but when i upgrade to v3, the inline script doesn't work anymore

here is my dynamic modal code 
``` 
<div x-data="{}" x-init="
        Livewire.on('closeModal', () => {
            $('#modal').modal('hide');
        });

        Livewire.on('openModal', () => {
            $('#modal').modal('show');
        });

        $('#modal').on('hidden.bs.modal', () => { @this.call('resetState'); document.body.style.overflowY = 'auto'; });
        $('#modal').on('shown.bs.modal', () => { document.body.style.overflowY = 'hidden'; }); ">
    <div wire:ignore.self class="modal fade" tabindex="-1" role="dialog" data-backdrop="static" id="modal">
        <div class="modal-dialog modal-dialog-centered align-items-md-start {{ $isFullScreen ? 'align-items-stretch' : '' }} {{ $modalSize }}"
            role="document">
            @if($component)
            @livewire($component, $data, key($component))
            @endif
        </div>
    </div>
</div>
```

i need to use inline script to run 3rd party library (like select2 for example)

any help is appreciated, thank you in advance.

**Comment by polycode-nz** [comment]
Any luck?

Similar problem with V3. 
The code is in the DOM but doesn't execute. A simple:

```
<script>
        console.log('test');
</script>
```

doesn't work. Nothing in console.

Tried `@teleport` too

```
@teleport('body')
    <script>
        console.log('test');
   </script>
@endteleport
  ```

**Reply by WireNinja** [reply]
Hi, @polycode-nz , if you need to execute js inside nested component, you can visit this discussions #6910 . Hope this help.

**Reply by polycode-nz** [reply]
Thanks @WireNinja I'll check it out. In my situation the js code is added to the DOM when a variable changes so it isn't there onload of the page. I thought that would be a common use case. CRUD with a modal.

If I add the js to the component view file it does work.

**Comment by wr0nggg** [comment]
Insted of @this use Livewire;
Livewire.find($(this).closest("div[wire\\:id]").attr("wire:id")).set('model', value);
